bridge/engine/EngineStreamDirector.h

```diff
@@ -873,11 +882,14 @@ class EngineStreamDirector
         // Unpinned, belogns to lastN and some of the participants needs this quality.
         for (const auto& participant : _participantStreams)
         {
-            if (participant.first != senderEndpointIdHash && participant.second.desiredUnpinnedLevel == quality)
+            if (participant.first != senderEndpointIdHash &&
```

Comment:
this comment is about line `877` (I can't add a comment on that line because it was not changed on this PR). The code is:
```
        if (anyParticipantsWithoutPinTarget())
        {
            return true;
        }
```

Could you explain to me why we return true if there is a participant without pin target?
It's seems to me if this quality is higher than `desiredUnpinnedLevel` in all participants, it will not be forwarder regardless whether a participant has a pin target or not

Reply:
There is a previous check: `if (!isSenderInLastNList)
        {
            return false;
        }`
Together it gives following rule:
_if current sender belongs to LastN set __AND__ there participants w/o pins_ - return `true`. (As if "yes, unpinned quality from this sender is indeed used).

In more details: participants w/o pin will receive _only_ videos from the LastN set, and only in non-high quality.

Reply:
Also, this rule: `if (!isSenderInLastNList) { return false; }` immediately filters out senders if they are not in the LastN. Theirs SSRC's still might be used, but as "pinned" and there is a separate check for this in the 'isSsrcUsed'.

Reply:
Yes, but if all participants are interested in lowQality, this will return `true` for midQuality. or if all want midQaulity, the lowQuality is not necessary.

Reply:
Fixed.